### DISCUSSION TITLE
API PUT request results in an error "{"error": "unknown_error"}".

### DISCUSSION THREAD

**Discussion by ko-morimoto** [keycloak/keycloak#22362] [Q&A]
Hello.

If you try to update the clientId of the SAML client with this command, you will get an error.

 curl -k -X PUT \
   -H "Content-Type: application/json" \
   -H "Authorization: Bearer {$TOKEN} \
   -d '{"clientId": hogehoge}' \
   "https://{$IPADDR}/admin/realms/master/clients/{$ID}"

=> {"error": "unknown_error"}

Running the same command against an OIDC client instead of SAML succeeds.
Items that are not clientId will not generate an error.
SAML will also succeed if you select an item other than clientId.

 curl -k -X PUT \
   -H "Content-Type: application/json" \
   -H "Authorization: Bearer {$TOKEN} \
   -d '{"name": hogehoge}' \
   "https://{$IPADDR}/admin/realms/master/clients/{$ID}"

=> OK

Why does the SAML clientId update fail?
The Keycloak version is 21.1.1.

Please give me some advice.
Thank you in advance.

**Comment by danielFesenmeyer** [comment]
In most cases, when you get a "unknown_error" result, you can find an ERROR in the logs. It would be helpful to get that error with stacktrace - otherwise others can just guess.
